lib/ferrum/browser.rb

```diff
@@ -77,6 +77,11 @@ def extensions
       end
     end
 
+    def evaluate_on_new_document(javascript = nil)
+      extensions << (block_given? ? yield : javascript)
```

Comment:
block is not needed I guess, and we definitely should accept something so not need for `= nil`

Reply:
Oh really? I liked that syntax:

```ruby
browser.evaluate_on_new_document do
  <<~JS
    Object.defineProperty(navigator, 'languages', {
      get: function() { return ['tlh']; }
    });
  JS
end
```

I changed it accordingly though.

Reply:
That syntax is almost still entirely valid, however by adding the block into the mix, there's a certain element of unnecessary indirection in the API. Questions that come to mind are "Does the block evaluate in a certain context? Are there args? How is this different from the `javascript` arg?"

You surely know this, but this is still valid:

```rb
browser.evaluate_on_new_document <<~JS
  Object.defineProperty(navigator, 'languages', {
    get: function() { return ['tlh']; }
  });
JS
```

And I leaves fewer questions when learning the API.

As a side note, I too love using the heredoc+highlight syntax for `js` strings. Particularly in Ferrum, it feels fantastic.

Reply:
> by adding the block into the mix, there's a certain element of unnecessary indirection in the API

For me blocks create a semantic boundary which goes further than "only" having a heredoc.
But again: I changed it anyway. :wink: